src/systems/bit-media-frames.js

```diff
@@ -133,89 +157,152 @@ const snapToFrame = (() => {
-  meshClone.traverse(node => {
-    updateMaterials(node, function (srcMat) {
-      const mat = srcMat.clone();
-      mat.transparent = true;
-      mat.opacity = 0.5;
-      mat.format = THREE.RGBAFormat;
-      mat.blending = THREE.NormalBlending;
-      node.material = mat;
-      return mat;
+  return (obj, scaleArray) => {
+    obj.updateMatrices();
+    obj.matrixWorld.decompose(position, quaternion, scale);
+    setMatrixWorld(obj, m4.compose(position, quaternion, scale.fromArray(scaleArray)));
+  };
+})();
+
+const frame2mixer = new Map();
+function createPreviewMesh(world, frame, capturable) {
+  let srcMesh;
+  let el;
+  let previewMesh;
+  let isVideo = false;
+  let ratio = 1;
+  if (hasComponent(world, AEntity, capturable)) {
+    el = world.eid2obj.get(capturable).el;
+    const video = el.components["media-video"];
+    isVideo = !!video;
+    if (isVideo) {
+      ratio =
+        (video.videoTexture.image.videoHeight || video.videoTexture.image.height) /
+        (video.videoTexture.image.videoWidth || video.videoTexture.image.width);
+    }
+    srcMesh = el.getObject3D("mesh");
+  } else {
+    const mediaEid = findChildWithComponent(world, MediaLoaded, capturable);
+    isVideo = hasComponent(world, MediaVideo, mediaEid);
+    if (isVideo) {
+      ratio = MediaVideo.ratio[mediaEid];
+    }
+    srcMesh = APP.world.eid2obj.get(mediaEid);
+  }
+
+  // Audios can't be cloned so we take a different path for them
+  if (isVideo) {
+    const previewMaterial = new THREE.MeshBasicMaterial();
+    previewMaterial.side = THREE.DoubleSide;
+    previewMaterial.transparent = true;
+    previewMaterial.opacity = 0.5;
+
+    const geometry = createPlaneBufferGeometry(1, 1, 1, 1, TEXTURES_FLIP_Y);
+    previewMesh = new THREE.Mesh(geometry, previewMaterial);
+    previewMesh.material.map = srcMesh.material.map;
+    previewMesh.material.needsUpdate = true;
+    // Preview mesh UVs are set to accommodate textureLoader default, but video textures don't match this
+    previewMesh.scale.y *= TEXTURES_FLIP_Y !== previewMesh.material.map.flipY ? -ratio : ratio;
+  } else {
+    previewMesh = cloneObject3D(srcMesh, false);
+    previewMesh.traverse(node => {
+      updateMaterials(node, function (srcMat) {
+        const mat = srcMat.clone();
+        mat.transparent = true;
+        mat.opacity = 0.5;
+        mat.format = THREE.RGBAFormat;
+        mat.blending = THREE.NormalBlending;
+        node.material = mat;
+        return mat;
+      });
     });
-  });
 
-  // TODO play animations on previews
+    if (hasComponent(world, AEntity, capturable)) {
+      const loopAnimation = el.components["loop-animation"];
+      if (loopAnimation && loopAnimation.isPlaying) {
+        const originalAnimation = loopAnimation.currentActions[loopAnimation.data.activeClipIndex];
+        const animation = previewMesh.animations[loopAnimation.data.activeClipIndex];
+        const mixer = new THREE.AnimationMixer(previewMesh);
```

Comment:
Probably we want to avoid `THREE` prefix. We may import Three.js objects from `three` and remove the prefix.